lib/matplotlib/axes/_base.py

```diff
@@ -1298,6 +1298,9 @@ def __clear(self):
         self.child_axes = []
         self._current_image = None  # strictly for pyplot via _sci, _gci
         self._projection_init = None  # strictly for pyplot.subplot
+
+        if hasattr(self, "legend_") and self.legend_ is not None:
+            self.legend_.axes = self.legend_.figure = None
```

Comment:
I wonder if this should be done in `remove_legend`, and then that called here?

Reply:
Happy with either, not sure if there's a reason to prefer one over the other. The only place `_remove_legend()` is currently called is as the callback for `Artist.remove()`, which unsets `self.figure` and `self.axes` already.

Reply:
Ah right, no benefit to moving it then.